DownFall/Tutorial/TutorialScene.swift

```diff
@@ -0,0 +1,220 @@
+        
+        // TutorialView
+        let tutorialView = TutorialView(tutorialData: GameScope.tutorialOne,
+                                        texture: nil,
+                                        color: .clear,
+                                        size:  CGSize(width: size.playableRect.width, height: size.playableRect.height))
+        tutorialView.position = .zero
+        tutorialView.zPosition = Precedence.menu.rawValue
+        
+        foreground.addChild(tutorialView)
+        
+        // Register for inputs we care about
+        Dispatch.shared.register { [weak self] input in
+            if input.type == .visitStore {
+                guard let self = self,
+                    let playerIndex = tileIndices(of: .player(.zero), in: self.board.tiles).first
+                    else { return }
+                
+                self.foreground.removeAllChildren()
+                if case let TileType.player(data) = self.board.tiles[playerIndex].type {
+                    let revivedData = data.revive()
+                    self.removeFromParent()
+                    self.gameSceneDelegate?.visitStore(revivedData)
+                }
+            }
+        }
+
+        //Turn watcher
+        TurnWatcher.shared.register()
+        
+        //Debug settings triple tap
+       let tapRecognizer = UITapGestureRecognizer(target: self, action: #selector(tripleTap))
+       tapRecognizer.numberOfTapsRequired = 3
+       view.addGestureRecognizer(tapRecognizer)
+    }
+    
+    public func prepareForReuse() {
+        board = nil
+        renderer = nil
+        foreground = nil
+        generator = nil
+        InputQueue.reset()
+        Dispatch.shared.reset()
+        print("deiniting")
+    }
+}
+
+//MARK: Swiping logic
+extension TutorialScene {
+    
+    @objc func swiped(_ gestureRecognizer: UISwipeGestureRecognizer) {
+        let inTop = isInTop(gestureRecognizer: gestureRecognizer)
+        let onRight = isOnRight(gestureRecognizer: gestureRecognizer)
+        touchWasSwipe = true
+        switch gestureRecognizer.direction {
+        case .down:
+            onRight ? rotateRight() : rotateLeft()
+        case .up:
+            !onRight ? rotateRight() : rotateLeft()
+        case .left:
+            !inTop ? rotateRight() : rotateLeft()
+        case .right:
+            inTop ? rotateRight() : rotateLeft()
+        default:
+            fatalError("There should only be four directions in our swipe gesture recognizer")
+        }
+    }
+    
+    private func isOnRight(gestureRecognizer: UISwipeGestureRecognizer) -> Bool {
+        let location = gestureRecognizer.location(in: self.view)
+        return location.x > (self.view?.frame.width ?? 0)/2
+    }
+    
+    private func isInTop(gestureRecognizer: UISwipeGestureRecognizer) -> Bool {
+        let location = gestureRecognizer.location(in: self.view)
+        return location.y < (self.view?.frame.height ?? 0)/2
+    }
+}
```

Comment:
This is copied and pasted, can we move it into an extension on UIView or SKView?